homeassistant/components/cookidoo/config_flow.py

```diff
@@ -0,0 +1,199 @@
+"""Config flow for Cookidoo integration."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from cookidoo_api import (
+    DEFAULT_COOKIDOO_CONFIG,
+    Cookidoo,
+    CookidooAuthException,
+    CookidooLocalizationConfig,
+    CookidooRequestException,
+    get_localization_options,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_NAME, CONF_PASSWORD
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import CONF_LOCALIZATION, DOMAIN, LOCALIZATION_SPLIT_CHAR
+from .coordinator import CookidooConfigEntry
+from .helpers import cookidoo_localization_for_key
+
+_LOGGER = logging.getLogger(__name__)
+
+AUTH_DATA_SCHEMA = {
+    vol.Required(CONF_EMAIL): TextSelector(
+        TextSelectorConfig(
+            type=TextSelectorType.EMAIL,
+            autocomplete="email",
+        ),
+    ),
+    vol.Required(CONF_PASSWORD): TextSelector(
+        TextSelectorConfig(
+            type=TextSelectorType.PASSWORD,
+            autocomplete="current-password",
+        ),
+    ),
+}
+
+
+class CookidooConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Cookidoo."""
+
+    VERSION = 1
+    reauth_entry: CookidooConfigEntry
+    reconfigure_entry: CookidooConfigEntry
+    AUTH_DATA_SCHEMA: dict
+    LOCALIZATION_DATA_SCHEMA: dict
+    country_codes: list[str]
+    languages: list[str]
+    localizations: list[CookidooLocalizationConfig]
```

Comment:
move those to instance instead of class vars

Reply:
Why and how? In did copy the `bring` setup which does have it like that? @zweckj

Reply:
how: just move inside init. why: imo it's good practice to only use class variables when ncessary, avoids weird side effects